app/components/dialogs/CreateLocationDialog.tsx

```diff
@@ -54,379 +55,433 @@ const styles: any = theme => ({
 });
 
 interface Props {
+  location?: Location;
   open: boolean;
   onClose: () => void;
   fullScreen: boolean;
-  addLocation: (location: Location) => void;
+  addLocation?: (location: Location) => void;
+  editLocation?: (location: Location) => void;
   showSelectDirectoryDialog: () => void;
 }
 
-interface State {
-  showAdvancedMode: boolean;
-  showSecretAccessKey: boolean;
-  errorTextPath: boolean;
-  errorTextName: boolean;
-  disableConfirmButton: boolean;
-  name: string;
-  path: string;
-  newuuid: string;
-  endpointURL: string;
-  perspective: string;
-  isDefault: boolean;
-  isReadOnly: boolean;
-  watchForChanges: boolean;
-  persistIndex: boolean;
-  fullTextIndex: boolean;
-  storeName: string;
-  cloudErrorTextName: boolean;
-  cloudErrorAccessKey: boolean;
-  cloudErrorSecretAccessKey: boolean;
-  cloudErrorBucketName: boolean;
-  cloudErrorRegion: boolean;
-  accessKeyId: string;
-  secretAccessKey: string;
-  bucketName: string;
-  region?: any;
-  storePath: string;
-  type: string;
-}
+const CreateLocationDialog = (props: Props) => {
+  const [showAdvancedMode, setShowAdvancedMode] = useState<boolean>(false);
+  const [showSecretAccessKey, setShowSecretAccessKey] = useState<boolean>(
+    false
+  );
+  const [errorTextPath, setErrorTextPath] = useState<boolean>(false);
+  const [errorTextName, setErrorTextName] = useState<boolean>(false);
+  // const [errorTextId, setErrorTextId] = useState<boolean>(false);
+  const [name, setName] = useState<string>(
+    props.location && props.location.type === locationType.TYPE_LOCAL
+      ? props.location.name
+      : ''
+  );
+  const [path, setPath] = useState<string>(
```

Comment:
I would like to support for some time the old way for saving path in the paths array. It should be done in addition to the new way for compatibility reasons.